scripts/poppunk_extract_distances.py

```diff
@@ -14,40 +15,60 @@ def get_options():
 
     # input options
     parser.add_argument('--distances', required=True, help='Prefix of input pickle and numpy file of pre-calculated distances (required)')
+    parser.add_argument('--tree', required=False, help='Newick file containing phylogeny of isolates', default = None)
     parser.add_argument('--output', required=True, help='Name of output file')
 
     return parser.parse_args()
 
-def iterDistRows(refSeqs, querySeqs, self=True):
+def listDistInts(refSeqs, querySeqs, self=True):
     """Gets the ref and query ID for each row of the distance matrix
-
     Returns an iterable with ref and query ID pairs by row.
-
     Args:
         refSeqs (list)
             List of reference sequence names.
         querySeqs (list)
             List of query sequence names.
         self (bool)
             Whether a self-comparison, used when constructing a database.
-
             Requires refSeqs == querySeqs
-
             Default is True
-        Returns:
-            ref, query (str, str)
-                Iterable of tuples with ref and query names for each distMat row.
-        """
+    Returns:
+        ref, query (str, str)
+            Iterable of tuples with ref and query names for each distMat row.
+    """
+    n = 0
+    num_ref = len(refSeqs)
+    num_query = len(querySeqs)
     if self:
-        if refSeqs != querySeqs:
-            raise RuntimeError('refSeqs must equal querySeqs for db building (self = true)')
-        for i, ref in enumerate(refSeqs):
-            for j in range(i + 1, len(refSeqs)):
-                yield(refSeqs[j], ref)
+        comparisons = [(0,0)] * int((num_ref * (num_ref-1)) * 0.5)
```

Comment:
Why remove the `yield` approach? This saves loading everything into memory
You can just replace:
```
yield(ref, query) -> yield(j, i)
```